### ISSUE TITLE
Why does izip!(iter) not return 1-tuples?

### ISSUE THREAD

**Issue by ExpHP** [bluss/rust-itertools#267]
When given a single Iterator, `izip` just does `into_iter`.

I know it has been like this from the very beginning, but I never understood why.  There isn't really much of a use case for calling izip with one iterable in real code ; the only time this ever happens is when another macro uses izip, in which case the lack of 1-tuples invariably just causes trouble.

Given some matched repetition like `$($ident)+`, I see no straightforward way to construct a pattern for these items.

```rust
#[macro_use] extern crate itertools;

macro_rules! fmt_columns {
    ($fmt:expr, $($c:ident),+ $(,)*) => {
        izip!($($c),+)
            // nope, matches A with (a,)
            // .map(|($($c,)+)| format!($fmt, $($c),+))
            
            // nope. (a) is not a valid pattern
            // .map(|($($c),+)| format!($fmt, $($c),+))
            
            // ???
    }
}
```

**Comment by bluss** [comment]
There's no why. Best to make the case for a change (or not) if it's wanted

**Comment by ExpHP** [comment]
Right.  The case for a change is because `izip!(a)` is not needed in handwritten code, but naturally arises in macros. 1-tuples compose better with macros because:
* it allows them to write `( $($a,)* )`.
* it eases the use of a trait as an implementation detail of a macro, because an impl involving `(A,)` (for all `A`) does not overlap with an impl involving `(A, B)` (for all `A, B`) or etc..  In contrast, an impl involving `A` tends to overlap with anything else.

**Comment by bluss** [comment]
This should have the breaking-change label so that we find it when doing breaking version updates.

@ExpHP You have a good argument. My only counter is that it's pretty easy to have macro code adapt to the unary case not being a tuple, at least in pattern matching, I've done that several times.
